java/ql/src/Security/CWE/CWE-501/TrustBoundaryViolation.qhelp

```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE qhelp PUBLIC
+ "-//Semmle//qhelp//EN"
+ "qhelp.dtd">
+<qhelp>
+  <overview>
+    <p>
+      A trust boundary violation occurs when a value is passed from a less trusted context to a more trusted context.
+    </p>
+    
+    <p>
+      For example, a value that is generated by a less trusted source, such as a user, may be passed to a more trusted
+      source, such as a system process. If the less trusted source is malicious, then the value may be crafted to
+      exploit the more trusted source.
+    </p>
+
+    <p>
+      Trust boundary violations are often caused by a failure to validate input. For example, if a web application
+      accepts a cookie from a user, then the application should validate the cookie before using it. If the cookie is
+      not validated, then the user may be able to craft a malicious cookie that exploits the application.
+    </p>
+  </overview>
+
+  <recommendation>
+    <p>
+      In order to maintain a trust boundary, data from less trusted sources should be validated before being used.
+    </p>
+  </recommendation>
+
+  <example>
+    <p>
+      In the first (bad) example, the server accepts a parameter from the user and uses it to set the username without validation.
+    </p>
+    <sample src="TrustBoundaryVulnerable.java" />
+
+    <p>
+      In the second (good) example, the server validates the parameter before using it to set the username.
```

Comment:
```suggestion
      In the second (good) example, the server validates the parameter from the user, then uses it to set the username.
```

Small suggestion to note again that the parameter comes from the user, just to be _extra_ clear 🙂 this helps maintain context for localization purposes.